.github/workflows/publish_pypi.yml

```diff
@@ -1,38 +1,63 @@
 name: Build and publish package to PyPI
 
 on:
-  push:
-    tags:
-    - '*'
+  release:
+    types: [published]
   workflow_dispatch:
     inputs:
       target:
-        description: 'Deployment target. Can be "pypi" or "testpypi"'
-        default: "pypi"
+        description: 'Deployment target. Can be "pypi" or "testpypi", or blank to skip publishing.'
+        default: ""
 
 jobs:
-  publish_pypi:
-    name: Build wheels on ubuntu-latest
+  build:
+    name: Build sdist and wheel
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: 3.8
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install build
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
```

Comment:
Not required if we use the action mentioned above.
Should specify the Python version if used (was a breaking change in `v4`).
```suggestion
```

Reply:
I'll drop it completely and use BAIPP

Reply:
Actually, I remembered the reason why I wouldn't like to use it: it doesn't pin the internal actions it uses (setup-python, upload-artifact, download-artifact) to their specific hashes, and I'd like to do that for this workflow because it is specific for a release. I'll add my own solution to inspecting the sdist and the wheel here, in case that helps!